codalab/codalab/settings/base.py

```diff
@@ -382,6 +372,12 @@ class Base(Settings):
     if isinstance(AWS_QUERYSTRING_AUTH, str) and 'false' in AWS_QUERYSTRING_AUTH.lower():
         AWS_QUERYSTRING_AUTH = False  # Was set to string, convert to bool
 
+    # Defaults to virtual because AWS recently deprecated 'path' style addressing. Boto3 required for virtual to have an effect.
+    AWS_S3_ADDRESSING_STYLE = os.environ.get('AWS_S3_ADDRESSING_STYLE', 'virtual')
+
+    # TODO: We will need to explicitly set this eventually. When set to None publicly uploaded files like images are not viewable. (Access Denied from AWS)
```

Comment:
is this taken care of?

Reply:
Not currently, I figured we'd need to discuss this option.

```
If set to private changes uploaded file’s Access Control List from the default permission public-read to give owner full control and remove read access from everyone else.
```

Might be an issue with my S3 bucket setup, but setting to `private` causes competition images, etc to not appear. Maybe it's possible to set it to private for only `BundleStorage` and have `PublicStorage` use `public-read`?

Reply:
Yeah you need to, maybe, inherit the base storage class and set some "public-read" permissions or some-such. I can try to find a reference for you if you need, maybe v2 does something similar?

Reply:
It looks like it sets an attribute on the Storage object itself:
https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py#L299

So it looks like we should be able to do:
```
PublicStorage.bucket_acl = 'public-read'
```

And have the setting default to `private`.